projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -139,6 +139,14 @@ export class IgxTimePickerComponent implements
             };
             this.onValueChanged.emit(args);
         } else {
+            const minTime = this.minValue ? this._convertMinMaxValue(this.minValue) : null;
```

Comment:
This looks like similar checks like `_isValueValid`, no? The issue with this is not in the value setter, outside/clear click doesn't fail like this going trough here as well.

Where it goes wrong is a combination of the `_onDropDownClosed` logic attempting to convert empty value (resulting in an Invalid Date) that passes as valid as is set to the editor, and likely the cancel action creating that empty value